### PR TITLE
UF+Arith: Enable substitutions at preprocessing

### PR DESCRIPTION
**Pull request by blishko**
We can run the same preprocessing of the formula as for the `UF` and `LA` theories.

It looks like the simplification code in `ArithLogic` can handle UF terms already, only a small fix was needed for the case where sum simplifies to an uninterpreted term.

### CODE REVIEW

File: src/logics/ArithLogic.cc [hunk 1]
```diff
@@ -372,11 +372,12 @@ PTRef ArithLogic::mkPlus(vec<PTRef> && args)
     args.clear();
     SymRef s_new;
     simp.simplify(plusSym, flattened_args, s_new, args);
-    if (args.size() == 1)
-        return args[0];
     if (s_new != getPlusForSort(returnSort)) {
         return mkFun(s_new, std::move(args));
     }
+    if (args.size() == 1) {
```
Comment (blishko) [inline src/logics/ArithLogic.cc:378]: Now I am thinking this might cause problems if sum simplifies to a multiplication?!

Comment (blishko) [reply src/logics/ArithLogic.cc:378]: No, it seems the code handles this case correctly.

**Comment by aehyvari** [comment]
Looks sufficiently innocent to me.

**Comment by blishko** [comment]
I ran QF_LRA benchmarks, and QF_UFLRA all benchmarks and model validation of SAT benchmarks, everything looks good, so I think this should be good to go.

**Review by aehyvari — APPROVED** [review]

**Pull request merged**